dist/src/main/java/io/camunda/zeebe/shared/security/IdentityAuthenticationManager.java

```diff
@@ -22,28 +26,47 @@
 public final class IdentityAuthenticationManager implements ReactiveAuthenticationManager {
 
   private final Identity identity;
+  private final MultiTenancyCfg multiTenancy;
 
   @Autowired
-  public IdentityAuthenticationManager(final Identity identity) {
+  public IdentityAuthenticationManager(
+      final Identity identity, final MultiTenancyCfg multiTenancy) {
     this.identity = identity;
+    this.multiTenancy = multiTenancy;
   }
 
   @Override
   public Mono<Authentication> authenticate(final Authentication authentication) {
-    if (!(authentication instanceof PreAuthToken preAuthToken)) {
+    if (!(authentication instanceof final PreAuthToken preAuthToken)) {
       return Mono.just(authentication);
     }
 
     final List<String> tenants;
     final var tokenValue = preAuthToken.token();
+    final AccessToken token;
 
     try {
-      final var token = identity.authentication().verifyToken(tokenValue);
-      tenants = Collections.singletonList(TenantOwned.DEFAULT_TENANT_IDENTIFIER);
-
-      return Mono.just(new IdentityAuthentication(token, tenants));
+      token = identity.authentication().verifyToken(tokenValue);
     } catch (final Exception e) {
       throw new BadCredentialsException(e.getMessage(), e);
     }
+
+    tenants = getTenants(tokenValue);
+
+    // TODO: if no tenants are returned should we set the default tenant?
```

Comment:
❓ Should we clarify this first or simply remove the comment because it's clear now?

Reply:
Ah right, we can remove this :+1: